services/apollo/index.ts

```diff
@@ -44,19 +44,20 @@ class Apollo {
 
   injectPoolData(pools: any[]): PoolData[] {
     const configPools = config.pools;
-    const tinlakePools = pools.map((pool: any) => {
-      const configPool = configPools.find(p => p.addresses.ROOT_CONTRACT === pool.id);
+    const tinlakePools = configPools.map((configPool: any) => {
+      const poolId = configPool.addresses.ROOT_CONTRACT;
+      const pool = pools.find(p => p.id === poolId);
       return {
-        id: pool.id,
-        name: configPool && configPool?.name || '',
-        asset: configPool && configPool?.asset || '',
-        ongoingLoans: pool.ongoingLoans.length, // TODO add count field to subgraph, inefficient to query all loans
-        totalDebt: new BN(pool.totalDebt),
-        totalRepaysAggregatedAmount: new BN(pool.totalRepaysAggregatedAmount),
-        weightedInterestRate: new BN(pool.weightedInterestRate),
-        weightedInterestRateDrop: new BN(0) // TODO how to get this value?
+        id: poolId,
+        name: configPool.name,
+        asset: configPool?.asset,
+        ongoingLoans: pool && pool.ongoingLoans.length || 0, // TODO add count field to subgraph, inefficient to query all loans
+        totalDebt: pool && new BN(pool.totalDebt) || new BN('0'),
+        totalRepaysAggregatedAmount: pool && new BN(pool.totalRepaysAggregatedAmount) || new BN('0'),
+        weightedInterestRate: pool && new BN(pool.weightedInterestRate) || new BN('0'),
+        weightedInterestRateDrop: new BN('0') // TODO how to get this value?
```

Comment:
This will be added in a future PR, right?

Reply:
yes :)